src/components/Medicine/MedicationAdministration/PrintMedicationAdministration.tsx

```diff
@@ -0,0 +1,493 @@
+import careConfig from "@careConfig";
+import { useQuery } from "@tanstack/react-query";
+import { format } from "date-fns";
+import { useMemo } from "react";
+import { useTranslation } from "react-i18next";
+import { formatPhoneNumberIntl } from "react-phone-number-input";
+
+import { cn } from "@/lib/utils";
+
+import PrintPreview from "@/CAREUI/misc/PrintPreview";
+
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "@/components/ui/table";
+
+import Loading from "@/components/Common/Loading";
+import PrintTable from "@/components/Common/PrintTable";
+
+import api from "@/Utils/request/api";
+import query from "@/Utils/request/query";
+import {
+  formatPatientAge,
+  getWeeklyIntervalsFromTodayTill,
+} from "@/Utils/utils";
+import { Encounter } from "@/types/emr/encounter";
+import { MedicationAdministrationRead } from "@/types/emr/medicationAdministration/medicationAdministration";
+import medicationAdministrationApi from "@/types/emr/medicationAdministration/medicationAdministrationApi";
+
+export const PrintMedicationAdministration = (props: {
+  facilityId: string;
+  encounterId: string;
+  patientId: string;
+}) => {
+  const { facilityId, encounterId, patientId } = props;
+  const { t } = useTranslation();
+
+  const { data: encounter } = useQuery<Encounter>({
+    queryKey: ["encounter", encounterId],
+    queryFn: query(api.encounter.get, {
+      pathParams: { id: encounterId },
+      queryParams: { facility: facilityId },
+    }),
+  });
+
+  const { data: medicationAdministrations, isLoading } = useQuery({
+    queryKey: ["medication_administrations", patientId],
+    queryFn: query.paginated(medicationAdministrationApi.list, {
+      pathParams: { patientId: patientId },
+      queryParams: {
+        encounter: encounterId,
+      },
+      pageSize: 1000,
+    }),
+    enabled: !!patientId,
+  });
+
+  const administrationsByAdministrator = useMemo(
```

Comment:
useMemo is a bit overkill here, no? 🤔  Should be relatively small amount of data since it's per encounter.

Reply:
LGTM overall though 👍